xpublish_edr/select.py

```diff
@@ -0,0 +1,137 @@
+        except KeyError as e:
+            raise ValueError(f"Invalid variable: {e}") from e
+
+    query_param_keys = list(query_params.keys())
+    for query_param in query_param_keys:
+        if query_param in edr_query_params:
+            del query_params[query_param]
+
+    method: Optional[str] = "nearest"
+
+    for key, value in query_params.items():
+        split_value = value.split("/")
+        if len(split_value) == 1:
+            continue
+        elif len(split_value) == 2:
+            query_params[key] = slice(split_value[0], split_value[1])
+            method = None
+        else:
+            raise ValueError(f"Too many values for selecting {key}")
+
+    ds = ds.sel(query_params, method=method)
+    return ds
+
+
+def select_postition(ds: xr.Dataset, point: shapely.Point) -> xr.Dataset:
+    """
+    Return a dataset with the position nearest to the given coordinates
+    """
+    if _is_regular_xy_coords(ds):
+        return _select_position_regular_xy_grid(ds, point)
+    else:
+        # TODO: Handle 2D coordinates
+        raise NotImplementedError("Only 1D coordinates are supported")
+
+
+def select_area(
+    ds: xr.Dataset,
+    polygon: shapely.Polygon,
+    buffer: float = 0.0001,
+) -> xr.Dataset:
+    """
+    Return a dataset with the area within the given polygon
+    """
+    if _is_regular_xy_coords(ds):
+        return _select_area_regular_xy_grid(ds, polygon, buffer)
+    else:
+        # TODO: Handle 2D coordinates
+        raise NotImplementedError("Only 1D coordinates are supported")
+
+
+def _coord_is_regular(da: xr.DataArray) -> bool:
+    """
+    Check if the DataArray has a regular grid
+    """
+    return len(da.shape) == 1 and da.name in da.dims
+
+
+def _is_regular_xy_coords(ds: xr.Dataset) -> bool:
+    """
+    Check if the dataset has 2D coordinates
+    """
+    return _coord_is_regular(ds.cf["X"]) and _coord_is_regular(ds.cf["Y"])
+
+
+def _select_position_regular_xy_grid(
+    ds: xr.Dataset,
+    point: shapely.Point,
+) -> xr.Dataset:
+    """
+    Return a dataset with the position nearest to the given coordinates
+    """
+    # Find the nearest X and Y coordinates to the point
+    return ds.cf.sel(X=point.x, Y=point.y, method="nearest")
+
+
+def _select_area_regular_xy_grid(
+    ds: xr.Dataset,
+    polygon: shapely.Polygon,
+    buffer: float = 0.0001,
+) -> xr.Dataset:
+    """
+    Return a dataset with the area within the given polygon
+    """
+    # To minimize performance impact, we first subset the dataset to the bounding box of the polygon
+    polygon = polygon.buffer(buffer)
+    (minx, miny, maxx, maxy) = polygon.bounds
+    ds = ds.cf.sel(X=slice(minx, maxx), Y=slice(maxy, miny))
+
+    # For a regular grid, we can create a meshgrid of the X and Y coordinates to create a spatial mask
+    pts = np.meshgrid(ds.cf["X"], ds.cf["Y"])
+
+    # Create a mask of the points within the polygon
+    mask = shapely.contains_xy(polygon, pts[0], pts[1])
```

Comment:
```suggestion
    mask = shapely.intersects_xy(polygon, pts[0], pts[1])
```

https://shapely.readthedocs.io/en/stable/reference/shapely.intersects.html#shapely.intersects
> Intersects implies that overlaps, touches, covers, or within are True.

do you have a test for this edge case?

```python
poly = shapely.Polygon.from_bounds(0, 1, 2, 3)
shapely.intersects_xy(poly, x=[0, 0.5, 1], y=[2, 2.5, 3])
# array([ True,  True,  True])
```

Reply:
yeah in `test_select.py`